src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy

```diff
@@ -333,4 +335,90 @@ class SwitchesSpec extends HealthCheckSpecification {
     }
+
+    @Tidy
+    @Tags(LOW_PRIORITY)
+    @Unroll
+    def "Able to partially update switch a 'location.#data.field' field"() {
+        given: "A switch"
+        def sw = topology.activeSwitches.first()
+        def initConf = northbound.getSwitch(sw.dpId)
+
+        when: "Request a switch partial update for a #data.field field"
+        def updateRequest = new SwitchPatchDto().tap {
+            it.location = new SwitchLocationDtoV2().tap { it."$data.field" = data.newValue }
+        }
+        def response = northboundV2.partialSwitchUpdate(sw.dpId, updateRequest)
+
+        then: "Update response reflects the changes"
+        response.location."$data.field" == data.newValue
+
+        and: "Changes actually took place"
+        northbound.getSwitch(sw.dpId).location."$data.field" == data.newValue
+
+        cleanup:
+        if (initConf.location."$data.field") {
+            northboundV2.partialSwitchUpdate(sw.dpId, new SwitchPatchDto().tap {
+                it.location = new SwitchLocationDtoV2().tap { it."$data.field" = initConf.location."$data.field" }
+            })
+        } else {
+            if (data.field in ["latitude", "longitude"]) {
+                northboundV2.partialSwitchUpdate(sw.dpId, new SwitchPatchDto().tap {
+                    it.location = new SwitchLocationDtoV2().tap { it."$data.field" = 0 }
+                })
+            } else {
+                northboundV2.partialSwitchUpdate(sw.dpId, new SwitchPatchDto().tap {
+                    it.location = new SwitchLocationDtoV2().tap { it."$data.field" = "" }
+                })
+            }
+        }
+
+        where:
+        data << [
+                [
+                        field   : "latitude",
+                        newValue: 654
+                ],
+                [
+                        field   : "longitude",
+                        newValue: 456
+                ],
+                [
+                        field   : "street",
+                        newValue: "testStreet"
+                ],
+                [
+                        field   : "city",
+                        newValue: "testCity"
+                ],
+                [
+                        field   : "country",
+                        newValue: "testCountry"
+                ]
+        ]
+    }
+
+    @Tidy
+    def "Able to partially update switch a 'pop' field"() {
+        given: "A switch"
+        def sw = topology.activeSwitches.first()
+        def initConf = northbound.getSwitch(sw.dpId)
+
+        when: "Request a switch partial update for a 'pop' field"
+        def newPopValue = "test_POP"
+        def response = northboundV2.partialSwitchUpdate(sw.dpId, new SwitchPatchDto().tap { it.pop = newPopValue })
+
+        then: "Update response reflects the changes"
+        response.pop == newPopValue
+
+        and: "Changes actually took place"
+        northbound.getSwitch(sw.dpId).pop == newPopValue
+
+        cleanup:
+        if (initConf.pop) {
```

Comment:
northboundV2.partialSwitchUpdate(sw.dpId, new SwitchPatchDto().tap { it.pop = initConf.pop ?: "" }